elixir/apps/domain/lib/domain/auth.ex

```diff
@@ -252,6 +260,37 @@ defmodule Domain.Auth do
     end)
   end
 
+  # Update default provider for client auth
+  def assign_default_provider(%Provider{} = provider, %Subject{} = subject) do
+    with :ok <- ensure_has_permissions(subject, Authorizer.manage_providers_permission()) do
+      {:ok, {:ok, default_provider}} =
+        Repo.transaction(fn ->
+          # 1. Clear default for all other providers
+          Provider.Query.not_disabled()
+          |> Provider.Query.by_account_id(provider.account_id)
```

Comment:
For safe measure, rather than:
```
|> Provider.Query.by_account_id(provider.account_id)
```

shouldn't we use:
```
|> Authorizer.for_subject(Provider, subject)
```

In theory, if you use `Provider.query.by_account_id` and a `subject` from another account is passed in, I believe this would still clear the default provider for the account the provider is in.

Reply:
Good catch, I'll add a test for that.

Reply:
Fixed